### ISSUE TITLE
CMAKE_LEGACY_CYGWIN_WIN32 Is Unset

### ISSUE THREAD

**Issue by Alexhuszagh** [libdynd/libdynd#1322]
DyND uses the WIN32 flag to remove the `-fPIC -Wno-ignored-attributes` flags during compilation, however, it fails to set the `CMAKE_LEGACY_CYGWIN_WIN32` flag at the top of CMakeLists.txt. This flag sets WIN32 to be true on MSYS2/Cygwin compilers, which do not support either flag (and would typically ignore them, however, `-Werror` is enabled).

The solution is to add the following line to the toplevel `CMakeLists.txt` at the top of the file:

```cmake
set(CMAKE_LEGACY_CYGWIN_WIN32 1)
```

Once again, I can submit a PR with the last 3 changes, if desired.

**Comment by insertinterestingnamehere** [comment]
Thanks for the report.
Good find.
If we can, I'd rather just do `if (WIN32 or CYGWIN)`, but it's not clear to me if that would handle the MSYS2 case you mentioned.
Any idea when exactly this is an issue within MSYS/MSYS2?
I haven't seen any issues using a CMake installed via the MSI file distributed by Kitware within MSYS2 myself. Maybe the issue shows up with binaries from elsewhere?

**Comment by Alexhuszagh** [comment]
@insertinterestingnamehere I don't believe it's an issue on most MSYS2 installs, only ones using a Cygwin-like environment (but those may not set `CYGWIN` to true). I've been using the CI service Tea-CI, which is what MSYS2 uses to test their project. Under those conditions, the build fails since `WIN32` is not set to be "truthy". Since this is a fairly rare case, but there are no side effects of `set(CMAKE_LEGACY_CYGWIN_WIN32 1)` for this project, I don't see any downside to defining it.

**Comment by insertinterestingnamehere** [comment]
Okay, it looks like the warning is coming from the msys2 package's patch for CMake. See https://github.com/Alexpux/MSYS2-packages/blob/master/cmake/cmake-3.6.0-msys.patch#L791. I'm not a huge fan of legacy anything, but this looks like a good solution here.
